### ISSUE TITLE
Cannot set Android icon when using custom Manifest

### ISSUE THREAD

**Issue by frei-0xff** [fyne-io/fyne#3077]
### Describe the bug:
Android application build by `fyne package -os android -release` installs with default green android icon and not the one set in FyneApp.toml
After inspecting apk file I found out that icon's PNG is properly added to `/res/mipmap-xxxhdpi-v4/icon.png`, but decoded AndroidManifest.xml looks like that:
```xml
<?xml version="1.0" encoding="utf-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android" android:versionCode="18" android:versionName="3.1.0" package="io.github.jacalz.test" platformBuildVersionCode="15" platformBuildVersionName="4.0.4-1406430">
    <uses-sdk android:minSdkVersion="15" android:targetSdkVersion="30"/>
    <application android:label="FyneTest" android:icon="0x7f020000" android:debuggable="false">
        <activity android:theme="@style/Theme" android:label="FyneTest" android:name="org.golang.app.GoNativeActivity" android:configChanges="keyboardHidden|orientation|uiMode">
            <meta-data android:name="android.app.lib_name" android:value="FyneTest"/>
            <intent-filter>
                <action android:name="android.intent.action.MAIN"/>
                <category android:name="android.intent.category.LAUNCHER"/>
            </intent-filter>
        </activity>
    </application>
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
    <uses-permission android:name="android.permission.INTERNET"/>
</manifest>
```
`android:icon="0x7f020000"` does not seem right.
### To Reproduce:
Steps to reproduce the behaviour:
1. Package any HelloWorld app by `fyne package -os android -release`
2. Install apk on Android phone

### Device (please complete the following information):
 - **OS:** Android
 - **Version:** 5.1.1
 - **Go version:** 1.18.3
 - **Fyne version:** 2.2.1

**Comment by andydotxyz** [comment]
Hmm, this has been working for years, I wonder what has happened.
Can you please help by checking if using the “-icon” parameter has the same result?

**Comment by frei-0xff** [comment]
@andydotxyz Checked. Yes, it is the same with `-icon` parameter.

**Comment by andydotxyz** [comment]
Can you share the icon you are using? Is it certainly a PNG?

**Comment by frei-0xff** [comment]
@andydotxyz Yes, of course it is. Here it is:
![Icon](https://user-images.githubusercontent.com/2990839/174437151-97c45bc5-08f0-48c4-aa9d-2e05742c5f3f.png)
Tried with standart Fyne's icon with grey cogwheel with the same result.
I don't think it can depend on icon itself. As I already poinitet out, the problem is in `android:icon` attribute in AndroidManifest.xml. It has to be something like `@mipmap/icon.png` but it is set to `0x7f020000`.

**Comment by infamous40** [comment]
> Hmm, this has been working for years, I wonder what has happened. Can you please help by checking if using the “-icon” parameter has the same result?

**Comment by andydotxyz** [comment]
Hmm, I cannot replicate this at all.
Can you please try packaging one of our provided apps and see if it installs as expected on your device?

**Comment by andydotxyz** [comment]
Also what is the result printed for `fyne version`?

**Comment by frei-0xff** [comment]
@andydotxyz fyne cmd is the latest `fyne cli version: v2.2.1`
Is there any already packaged apk file that is known to show the right icon, so I can check it on my phone?
Maybe the problem is in old Android version on my device or something like that.

**Comment by frei-0xff** [comment]
@andydotxyz Tried to install Calculator and FyneGameBoy. They both installed with the default green icon. And won't
run at all. Thus, it is an incompatibility problem, probably because of old Android version 5.1.1 on my
device.

**Comment by andydotxyz** [comment]
Thanks for this. It should work on 5.1.1 though, I tested on 4.1.
Is there anything helpful in the log when they crash and/or load perhaps relating to icon or otherwise.

**Comment by frei-0xff** [comment]
@andydotxyz How can I see the log on Android phone?

**Comment by andydotxyz** [comment]
Use `adb logcat`

**Comment by frei-0xff** [comment]
@andydotxyz Sorry, I don't have Android SDK installed. I used fyne-cross containers for packaging apk.

**Comment by andydotxyz** [comment]
> I used fyne-cross containers for packaging apk.

Ah, this is crucial information. The `fyne` version of your system is not relevant like I had thought. What version of `fyne-cross` are you using?

**Comment by frei-0xff** [comment]
@andydotxyz I understand that. As I said, I use the latest version for packaging. I use cross-fyne's container `fyneio/fyne-cross:1.2-android` as environment for packaging. And before packaging I update fyne command to the latest version by `go install fyne.io/fyne/v2/cmd/fyne@latest`

**Comment by frei-0xff** [comment]
> You are not using the fyne-cross command itself?
No, I'm running command `fyne package -os android -release` inside Docker container `fyneio/fyne-cross:1.2-android` that has preinstalled environment for building Android application.
It's just easier than installing all of that on my local computer.

**Comment by andydotxyz** [comment]
Hmm, thanks.

Is this a standard Android device? What make/model is the 5.1.1 OS running on?

**Comment by frei-0xff** [comment]
> Is this a standard Android device? What make/model is the 5.1.1 OS running on?

Yes it's Huawei GR5. It is quite old.

**Comment by snonux** [comment]
I am having the same issue. However, the App is falling back to the default icon *only* when I have an `AndroidManifest.xml` and then run `fyne package --os android --icon icon.png`. Without any `AndroidManifest.xml` the icon just show sup correctly (but then I can't use the `AndroidManifest.xml` to request/set app storage permissions).

 When I try to manually set the icon in the `AndroidManifest.xml` I get: "manual declaration of android:icon in AndroidManifest.xml not supported"

My manifest currently looks like this:

```
<?xml version="1.0" encoding="utf-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
   package="org.buetow.quicklogger">

   <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
   <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
   <uses-permission android:name="android.permission.MANAGE_EXTERNAL_STORAGE" />

   <application
       android:allowBackup="true"
       android:label="quicklogger"
       android:icon="icon">

        <activity android:name="org.golang.app.GoNativeActivity">
            <meta-data android:name="android.app.lib_name" android:value="npc" />
            <intent-filter>
                <action android:name="android.intent.action.MAIN"/>
                <category android:name="android.intent.category.LAUNCHER"/>
            </intent-filter>
        </activity>
   </application>
</manifest>
```
I am using:

```
fyne version v2.4.3
```

**Comment by andydotxyz** [comment]
> I am having the same issue. However, the App is falling back to the default icon _only_ when I have an `AndroidManifest.xml` and then run `fyne package --os android --icon icon.png`. Without any `AndroidManifest.xml` the icon just show sup correctly (but then I can't use the `AndroidManifest.xml` to request/set app storage permissions).
> 
> When I try to manually set the icon in the `AndroidManifest.xml` I get: "manual declaration of android:icon in AndroidManifest.xml not supported"
> 
> My manifest currently looks like this:
> 
> ```
> <?xml version="1.0" encoding="utf-8"?>
> <manifest xmlns:android="http://schemas.android.com/apk/res/android"
>    package="org.buetow.quicklogger">
> 
>    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
>    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
>    <uses-permission android:name="android.permission.MANAGE_EXTERNAL_STORAGE" />
> 
>    <application
>        android:allowBackup="true"
>        android:label="quicklogger"
>        android:icon="icon">
> 
>         <activity android:name="org.golang.app.GoNativeActivity">
>             <meta-data android:name="android.app.lib_name" android:value="npc" />
>             <intent-filter>
>                 <action android:name="android.intent.action.MAIN"/>
>                 <category android:name="android.intent.category.LAUNCHER"/>
>             </intent-filter>
>         </activity>
>    </application>
> </manifest>
> ```
> 
> I am using:
> 
> ```
> fyne version v2.4.3
> ```

The icon reference will have to match the lookup generated in the compiled assets.
I recommend generating an app without a custom manifest, then extracting it and making the changes you want, then running again. It should work.

**Comment by snonux** [comment]
Thanks, but that didn't work for me. When I extract the `AndroidManifest.xml` from the apk file and use that one I get still "manual declaration of android:icon in AndroidManifest.xml not supported".

What worked however was to clone the fyne project, change the occurances of `android.permission.INTERNET` to `android.permission.MANAGE_INTERNAL_STORAGE` (as these are all the changes I want in my custom android manifest) in the source tree, re-compile fyne. Now all works. I know, this is more of an hammer approach!

**Comment by andydotxyz** [comment]
Updated the title to report the real issue.
I think the solution may be to support generically adding permission /role requests in the app metadata which we can add to the platform specific formats when packaging.

**Comment by snonux** [comment]
Generic support for adding permissions in the app metadata would be good, indeed

**Comment by andydotxyz** [comment]
Please report issues with Fyne-cross in that project, the build command for Fyne is separate and processes metadata in a different way.

**Comment by snonux** [comment]
Actually, this works now for me when using `fyne-cross android`

**Comment by andydotxyz** [comment]
Excellent, should this issue be closed then?

**Comment by snonux** [comment]
For me personally, yes. Don't know what the others, who commented on this issue, say. :-) 

Many thanks!
